internal/worker/rundynamic.go

```diff
@@ -189,7 +189,7 @@ func RunDynamicAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, sbOpts []sandb
 		phaseCtx := log.ContextWithAttrs(ctx, log.LabelAttr("phase", string(phase)))
 		startTime := time.Now()
 		args := dynamicanalysis.MakeAnalysisArgs(pkg, phase)
-		phaseResult, err := dynamicanalysis.Run(sb, analysisCmd, args, log.GetDefaultLogger())
+		phaseResult, err := dynamicanalysis.Run(ctx, sb, analysisCmd, args)
```

Comment:
Can we keep the logger argument here? I think the logger is better overridden in the worker package (which is called directly from the main files in `cmd` rather than the dynamic analysis package which is not called directly

Reply:
I'm a bit confused by what you're suggesting here? Do you mean propagate the logger just for strace separately here or all the internal logging for dynamicanalysis?

Would it better to wire up a featureflag to enable/disable strace logging?

Reply:
For strace separately, so that it can be redirected to a separate file.
We can also make a feature flag to disable strace logging but that's a separate thing (which incidentally, can be implemented by passing a nop logger in this argument)